pkg/runtime/bulk_subscriber.go

```diff
@@ -646,32 +688,32 @@ func endSpans(spans []trace.Span) {
 
 // sendBulkToDeadLetter sends the bulk message to deadletter topic.
 func (a *DaprRuntime) sendBulkToDeadLetter(
-	ctx context.Context, name string, msg *pubsub.BulkMessage, deadLetterTopic string,
-	entryIdIndexMap *map[string]int, bulkResponses *[]pubsub.BulkSubscribeResponseEntry, //nolint:stylecheck
-	bulkSubDiag *bulkSubIngressDiagnostics,
+	bulkSubCallData *bulkSubscribeCallData, msg *pubsub.BulkMessage, deadLetterTopic string,
+	sendAllEntries bool,
 ) error {
+	bscData := *bulkSubCallData
 	data := make([]pubsub.BulkMessageEntry, len(msg.Entries))
 
-	if bulkResponses == nil {
+	if sendAllEntries {
 		data = msg.Entries
 	} else {
 		n := 0
 		for _, message := range msg.Entries {
-			entryId := (*entryIdIndexMap)[message.EntryId] //nolint:stylecheck
-			if (*bulkResponses)[entryId].Error != nil {
+			entryId := (*bscData.entryIdIndexMap)[message.EntryId] //nolint:stylecheck
```

Comment:
can there be scenario that the entryId is not present in the map?

Reply:
No, the way it can reach this code-piece, entryIdIndexMap will have those entries.
That is taken care of separately when entries are sent by App, and if they are outside the known entries.